client/state/utils/serialize.ts

```diff
@@ -0,0 +1,35 @@
+/**
+ * External dependencies
+ */
+import { getInitialState } from '@automattic/state-utils';
+import type { Reducer, AnyAction } from 'redux';
+
+export type SerializableReducer< TState, TAction extends AnyAction > = Reducer<
+	TState,
+	TAction
+> & {
+	serialize?: ( state: TState ) => any;
+	deserialize?: ( persisted: any ) => TState;
+};
+
+export function serialize< TState, TAction extends AnyAction >(
+	reducer: SerializableReducer< TState, TAction >,
+	state: TState
+): any {
+	if ( ! reducer.serialize ) {
+		return undefined;
+	}
+
+	return reducer.serialize( state );
+}
+
+export function deserialize< TState, TAction extends AnyAction >(
+	reducer: SerializableReducer< TState, TAction >,
+	persisted: any
```

Comment:
should we use `unknown` rather than `any` throughout this file?

Reply:
Not sure. It should just match the return type of `serialize`. Maybe @jsnajdr can give more insight here.

Reply:
I don't understand the differences between `unknown` and `any` very well. But the native APIs like `JSON.parse` or `IDBObjectStore.get` return `any`, and `serialize` and `deserialize` functions are meant to work seamlessly with these native storage APIs.